pkg/controller/instance/instance_controller.go

```diff
@@ -271,7 +271,11 @@ func updateInstance(instance *kudov1beta1.Instance, oldInstance *kudov1beta1.Ins
 
 	// 1. check if the finalizer can be removed (if the instance is being deleted and cleanup is completed) and then
 	// update instance spec and metadata. this will not update Instance.Status field
-	isDelete := tryRemoveFinalizer(instance)
+	isDeletionComplete := false
```

Comment:
Still misleading IMHO, because the _deletion_ is complete when the Instance is removed, which is not the case here. We could call it what it is: `isFinalizerRemoved` (1) or what it means to us: `isCleanupDone` (2). I'd prefer the first one.

Reply:
Actually, the result from tryRemoveFinalizer is not important for us, that could also happend on an upgrade if the cleanup plan is removed. I've pushed another version...